src/backend/distributed/deparser/deparse_database_stmts.c

```diff
@@ -205,3 +249,112 @@ DeparseAlterDatabaseSetStmt(Node *node)
 
 	return str.data;
 }
+
+
+/*
+ *  Validates for if option is template, lc_type, locale or lc_collate, propagation will
+ * not be supported since template and strategy options are not stored in the catalog
+ * and lc_type, locale and lc_collate options depends on template parameter.
+ */
+static void
+ValidateCreateDatabaseOptions(DefElem *option)
+{
+	if (strcmp(option->defname, "template") == 0 ||
+		strcmp(option->defname, "strategy") == 0 ||
+		strcmp(option->defname, "lc_ctype") == 0 ||
+		strcmp(option->defname, "locale") == 0 ||
+		strcmp(option->defname, "lc_collate") == 0 ||
+		strcmp(option->defname, "icu_locale") == 0 ||
+		strcmp(option->defname, "locale_provider") == 0)
```

Comment:
As proposed over chat by @onurctirtir let's only disallow `template`. The others should still be allowed if you explicitely specify the value that the `template1` database has.

Reply:
You mean ` template0` I think since there is a different locale set other than `template1`
In this case, I need to create a table like `pg_dist_database`. I need to save template and strategy in this table and when propagating in metadata_sync in citus_add_node, I can get the information there. Is it the design what you mean?
@onurctirtir @JelteF

Reply:
No, I meant `template1`. `template1` is the one that's used by default to create a database with. We should still support the user explicitely specifying the same values that `template1` has.

If `template1` has UTF8 encoding we should still support a user running the following:
```
CREATE DATABASE mytestdb ENCODING 'UTF8';
```

This is easy to do, by removing all these checks except for the one for `"template"`.

And actually while thinking about this I think we should also still support specifying the template as `template1` or default. So these commands should also work:
```
CREATE DATABASE db_based_on_template1 TEMPLATE 'template1';
CREATE DATABASE db_based_on_default_template TEMPLATE DEFAULT;
```

Reply:
I see. But we can not support Strategy in either case since it is not stored. I will add a check for template which will accept  template1 or DEFAULT. Is it that what you mean? Because I can not totally remove the template check. It can not be `template0` or any other template other than `template1` or `DEFAULT

Reply:
> I will add a check for template which will accept template1 or DEFAULT. Is it that what you mean? Because I can not totally remove the template check. It can not be `template0` or any other template other than `template1` or `DEFAULT

Sounds good.

Regarding `STRATEGY`, it makes sense what you say for the general case. But based on the Postgres docs I don't think it really matters if we use a different value on node scale out. The final result of the CREATE DATABASE command should be the same no matter what STRATEGY is used. It really only impacts the speed of creating the database. So I think we can accept the STRATEGY argument, but only actually use the user provided value when the initial CREATE DATABASE command is executed, and then don't provide it when we run the CREATE DATABASE command during addition of a new node.

Reply:
Implemented with only accepting strategy with wal_log value